lumicks/pylake/file.py

```diff
@@ -160,26 +183,29 @@ def _get_photon_count(self, name):
     def _get_photon_time_tags(self, name):
         return TimeTags.from_dataset(self.h5["Photon Time Tags"][name], "Photon time tags")
 
+    def _get_object_dictionary(self, field, cls):
+        if field not in self.h5:
+            return dict()
+        return {name: cls.from_dataset(dset, self) for name, dset in self.h5[field].items()}
+
     @property
     def kymos(self) -> Dict[str, Kymo]:
-        if "Kymograph" not in self.h5:
-            return dict()
-        return {name: Kymo.from_dataset(dset, self) for name, dset in self.h5["Kymograph"].items()}
+        return self._get_object_dictionary("Kymograph", Kymo)
 
     @property
     def point_scans(self) -> Dict[str, Scan]:
-        if "Point Scan" not in self.h5:
-            return dict()
-        return {name: PointScan(dset, self) for name, dset in self.h5["Point Scan"].items()}
+        return self._get_object_dictionary("Point Scan", PointScan)
 
     @property
     def scans(self) -> Dict[str, Scan]:
-        if "Scan" not in self.h5:
-            return dict()
-        return {name: Scan.from_dataset(dset, self) for name, dset in self.h5["Scan"].items()}
+        return self._get_object_dictionary("Scan", Scan)
 
     @property
     def fdcurves(self) -> Dict[str, FDCurve]:
-        if "FD Curve" not in self.h5:
+        return self._get_object_dictionary("FD Curve", FDCurve)
+
+    @property
+    def markers(self) -> Dict[str, dict]:
+        if "Marker" not in self.h5:
             return dict()
-        return {name: FDCurve.from_dset(dset, self) for name, dset in self.h5["FD Curve"].items()}
+        return {name: Marker.from_dataset(dset) for name, dset in self.h5["Marker"].items()}
```

Comment:
It would be nice to also make the `Marker.from_dataset()` consistent with the rest and use `_get_object_dictionary()`.

Reply:
It did cross my mind, but having a dummy argument felt very icky. I wasn't 100% sure which one was the road of least groan in this case:

```    @staticmethod
    def from_dataset(h5py_dset, file):
        """
        Construct Marker class from dataset.

        Parameters
        ----------
        h5py_dset : h5py.Dataset
            The original HDF5 dataset containing Marker information
        file : lumicks.pylake.File
            The parent file. Unused, but part of the interface.
        """
        del file
        return Marker(h5py_dset.attrs)```

Reply:
I would recommend that you add a `.file` attribute to `Marker`. That makes it consistent with the other types. Even if it's not used internally, it's nice to have as an attribute so that you can always trace back where a certain item came from.